dev-client/src/App.tsx

```diff
@@ -78,18 +79,22 @@ function App(): React.JSX.Element {
   return (
     <GestureHandlerRootView style={style}>
       <Provider store={store}>
-        <NativeBaseProvider theme={theme}>
-          <Portal.Host>
-            <NavigationContainer>
-              <GeospatialProvider>
-                <Toasts />
-                <HomeScreenContextProvider>
-                  <RootNavigator />
-                </HomeScreenContextProvider>
-              </GeospatialProvider>
-            </NavigationContainer>
-          </Portal.Host>
-        </NativeBaseProvider>
+        <BottomSheetModalProvider>
+          <NativeBaseProvider theme={theme}>
+            <Portal.Host>
+              <NavigationContainer>
+                <BottomSheetModalProvider>
```

Comment:
We need this provider to exist below the `NavigationContainer` and `NativebaseProvider` because all other bottom sheets need to render on top of NativeBase floating elements (e.g., the Floating Action Button / FAB) and also may need access to the navigation context to allow buttons to navigate the user to different parts of the app.